src/App.tsx

```diff
@@ -1,54 +1,100 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  // const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isfocused, setIsfocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person>();
+  const [isListVisible, setIsListVisible] = useState(true);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer
+      .filter(person => person.name.toLowerCase().includes(
+        appliedQuery.toLowerCase(),
+      ));
+  }, [appliedQuery]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce((value) => {
+    setAppliedQuery(value);
+    setIsListVisible(true);
+  }, 1000),
+  []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setIsListVisible(false);
+    applyQuery(event.target.value);
+  };
+
+  const handleClick = (name: string) => {
+    setQuery(name);
+
+    const filterdPerson: Person | undefined
+      = peopleFromServer.find(person => person.name === name);
+
+    setSelectedPerson(filterdPerson);
+  };
+
+  const handleMouseDown = (
+    (event: React.KeyboardEvent<HTMLDivElement>, name: string) => {
+      if (event.key === 'Enter') {
+        handleClick(name);
+      }
+    });
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {selectedPerson
+        ? (
+          <h1 className="title">
+            {`${selectedPerson.name} (${selectedPerson.born} = ${selectedPerson.died})`}
+          </h1>
+        )
+        : (<h1 className="title">No selected person</h1>)}
```

Comment:
In order not to duplicate one tag with the same class twice, the check must be entered inside the tag